arch/arm/core/cortex_m/pm_s2ram.S

```diff
@@ -99,8 +99,20 @@ SECTION_FUNC(TEXT, arch_pm_s2ram_suspend)
 	bx	lr
 
 
+#if defined(CONFIG_PM_S2RAM_CUSTOM_BOOTSTACK)
+GTEXT(soc_pm_s2ram_custom_bootstack)
+#endif
+
 GTEXT(arch_pm_s2ram_resume)
 SECTION_FUNC(TEXT, arch_pm_s2ram_resume)
+
+#if defined(CONFIG_PM_S2RAM_CUSTOM_BOOTSTACK)
+	/* Setup stack for soc specific initialisation after wake up,
+	 * so the main stack does not get corrupted in the process.
+	 */
+	bl soc_pm_s2ram_custom_bootstack
```

Comment:
This branch overwrites the LR value which keeps the return address from `arch_pm_s2ram_resume` (see `bx lr` instruction below which is executed on normal boot). Thus no valid return from `arch_pm_s2ram_resume` as we can't store it on the stack.  The [previous ](https://github.com/zephyrproject-rtos/zephyr/commit/2ac567c30a61051427e038e168b746184945dcf3#diff-9857d024b8bfb36a683fed7b2782cfa800c0d2645914df02a6c3d0e067f1f764R47) version (call from `reset.S`) did not have this problem.